src/main/java/duke/DukeCommandValidator.java

```diff
@@ -0,0 +1,147 @@
+package duke;
+
+public class DukeCommandValidator {
+    static int MINIMUM_DONE_LENGTH = 5;
+    static int MINIMUN_TODO_LENGTH = 6;
+    static int MINIMUM_EVENT_LENGTH = 12;
+    static int MINIMUM_DEADLINE_LENGTH = 15;
+    static int MINIMUM_DELETE_LENGTH = 8;
+
+    public static int getCommand(String input){
+        if(input.equals("bye")){
+            return DukeCommands.EXIT;
+        }
+        if(input.equals("list")){
+            return DukeCommands.LIST;
+        }
+        if(isDoneCommandValid(input) == true){
+            return DukeCommands.DONE;
+        }
+        if(isAddCommandValid(input) == true) {
+            return DukeCommands.ADD;
+        }
+        if(isDeleteCommandValid(input) == true) {
+            return DukeCommands.DELETE;
+        }
+        return DukeCommands.INVALID_COMMAND;
+    }
+
+    public static boolean isToDoValid(String input){
+        if(input.length()>=MINIMUN_TODO_LENGTH) {
+            if (input.substring(0, 5).equals("todo ")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean isDeleteCommandValid(String input){
+        if(input.length()<MINIMUM_DELETE_LENGTH){
+            return false;
+        }
+        if(!input.substring(0,7).equals("delete ")){
+            return false;
+        }
+        // valid string after "delete "
+        String indexToDelete = input.substring(7);
```

Comment:
I'd suggest that it's best to avoid "magic numbers" where possible; perhaps you could use static final constants to define these numbers beforehand, so that a reader would know what these numbers are supposed to represent.